From f85df6a50b8ff09109e00d63cbe2dc449d3d714d Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Thu, 1 Nov 2018 15:01:06 -0500
Subject: [PATCH 4972/5242] MLK-20170: mtd: flexspi: change the flexspi dummy
 pad settings

commit  26fbb4b05d66d86165102676e15d824a4f17debf from
https://source.codeaurora.org/external/imx/linux-imx.git

The dummy pad settings should be align with data access mode, such as
set to PAD4 for all Quad read.

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/spi-nor/fsl-flexspi.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mtd/spi-nor/fsl-flexspi.c b/drivers/mtd/spi-nor/fsl-flexspi.c
index 2bd1580..c7b4eae 100644
--- a/drivers/mtd/spi-nor/fsl-flexspi.c
+++ b/drivers/mtd/spi-nor/fsl-flexspi.c
@@ -547,7 +547,7 @@ static void fsl_flexspi_init_lut(struct fsl_flexspi *flex)
 		writel(LUT0(CMD, PAD1, op) | LUT1(ADDR, PAD1, addrlen),
 		       base + FLEXSPI_LUT(lut_base));
 
-		writel(LUT0(DUMMY, PAD1, dm) |
+		writel(LUT0(DUMMY, PAD4, dm) |
 		       LUT1(FSL_READ, PAD4, 0),
 		       base + FLEXSPI_LUT(lut_base + 1));
 	/* DDR Quad I/O Read 	 */
@@ -558,7 +558,7 @@ static void fsl_flexspi_init_lut(struct fsl_flexspi *flex)
 		       base + FLEXSPI_LUT(lut_base));
 
 		writel(LUT0(MODE_DDR, PAD4, 0xff) |
-		       LUT1(DUMMY, PAD1, dm),
+		       LUT1(DUMMY, PAD4, dm),
 		       base + FLEXSPI_LUT(lut_base + 1));
 
 		writel(LUT0(READ_DDR, PAD4, 0) |
-- 
1.7.9.5

